ush/ufsda/stage.py

```diff
@@ -303,6 +317,30 @@ def gdas_single_cycle(config):
             target_file = target_file.replace('nc4', 'txt')
             r2d2_config['target_file_fmt'] = target_file
             ufsda.r2d2.fetch(r2d2_config)
+    # if hybvar, link ensemble members to DATA
+    if config['DOHYBVAR']:
+        for imem in range(1, config['NMEM_ENKF']+1):
+            memchar = f"mem{imem:03d}"
+            print(f"background_ens:  stage {memchar}")
```

Comment:
Do you want print statements or should this be outputted to a logger file?

Reply:
Good suggestion.  `print` was replaced with `logging.info` at 0ba33d8.

Much to my surprise, no output is generated by `logging.info` when `run_jedi_exe.py`   I added a `print` statement before `logging.info` and only the `print` was echoed to stdout.

A check of `ush/run_jedi_exe.py` shows other instances of `logging.info` which do not generate printout when the script is executed.   This is odd because `run_jedi_exe.py` sets the logging level to `INFO`
```
def run_jedi_exe(yamlconfig):
    logging.basicConfig(format='%(asctime)s:%(levelname)s:%(message)s', level=logging.INFO, datefmt='%Y-%m-%d %H:%M:%S')
    # open YAML file to get config
    try:
        with open(yamlconfig, 'r') as yamlconfig_opened:
            all_config_dict = yaml.safe_load(yamlconfig_opened)
        logging.info(f'Loading configuration from {yamlconfig}')
```
When `run_jedi_exe.py` is executed, the `Loading configuration from ...` message is _not_ printed.   The only printout I see when executing `run_jedi_exe.py` is from R2D2.   I do not understand why we see this behavior.   Any ideas @CoryMartin-NOAA?

Reply:
As a test, I commented out 
```
from ufsda.genYAML import genYAML
```
at the top of `run_jedi_exe.py`.    With this line commented out, `run_jedi_exe.py` generated info log messages up to the point where the script referenced `genYAML`.   

```
(gdasapp) Hera(hfe05):/scratch1/NCEPDEV/da/Russ.Treadon/git/global-workflow/test2/sorc/gdas.cd/ush$ ./run_jedi_exe.py -c examples/run_jedi_exe/3denvar_hera.yaml
2023-03-25 20:50:42:INFO:Loading configuration from examples/run_jedi_exe/3denvar_hera.yaml
2023-03-25 20:50:42:INFO:Created working directory /scratch2/NCEPDEV/stmp1/Russ.Treadon/gdas_single_test_3denvar.case_anl                              _c48
2023-03-25 20:50:44:INFO:Note: detected 80 virtual cores but NumExpr set to maximum of 64, check "NUMEXPR_MAX_THREADS" en                              vironment variable.
2023-03-25 20:50:44:INFO:Note: NumExpr detected 80 cores but "NUMEXPR_MAX_THREADS" not set, so enforcing safe limit of 8.
2023-03-25 20:50:57:INFO:Using yamlconfig examples/run_jedi_exe/3denvar_hera.yaml
Traceback (most recent call last):
  File "./run_jedi_exe.py", line 177, in <module>
    run_jedi_exe(args.config)
  File "./run_jedi_exe.py", line 139, in run_jedi_exe
    genYAML(yamlconfig, output=output_file)
NameError: name 'genYAML' is not defined
```

Why does importing `genYAML` from `ufsda.genYAML` turn off info level logging?

Reply:
The following `import` section is inside `def run_jedi_exe(yamlconfig):` in `run_jedi_exe.py`
```
    # add your ufsda python package to path
    ufsda_path = os.path.join(all_config_dict['GDASApp home'], 'ush')
    sys.path.append(ufsda_path)
    import ufsda
    from ufsda.misc_utils import calc_fcst_steps
    from ufsda.stage import gdas_single_cycle, gdas_fix, background_ens, atm_obs, bias_obs
```
I removed `from ufsda.genYAML import genYAM` from the top of `run_jedi_exe.py` and added it to the above `import` section.    With this change, `run_jedi_exe.py` runs to completion with info messages written to stdout.    This change was committed at [`f57fae0`](https://github.com/RussTreadon-NOAA/GDASApp/commit/f57fae0199ccdf2ff863623cb52ff9cb97f99210)

I don't understand why this change yields the desired behavior.

Reply:
Is it possible it was using a different version of `genYAML` when it was imported before the `sys.path.append` section? I wonder if that has anything to do with it. Otherwise this is very strange indeed.

Reply:
Various tests this morning adding `print` to see what's loaded or moving `import` around did not uncover any explanation for the observed behavior.   `run_jedi_exe.py` at [`f57fae0`](https://github.com/RussTreadon-NOAA/GDASApp/commit/f57fae0199ccdf2ff863623cb52ff9cb97f99210) runs with log statements correctly written to stdout.